### PR TITLE
migrate entity pages to use relative routes for react-router v6 stable compatibility

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

More work towards #9913, on top of #10450

This migrates the absolute entity paths to be relative, which we'll for react-router v6 stable

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by freben — APPROVED** [review]
Nice. And this is working with rr-dom beta too?

**Comment by Rugvip** [comment]
@freben yep, although will do another round of verification once all builds are green. Pretty much all of the `beta.0` -> stable changes seem to be possible to express in ways that are compatible with both, so I think that's how we'll do the migration. Migrations, deprecations, and optional version bump going out in 1.1, and then in some later release we drop support for `beta.0`

File: packages/app/src/App.tsx [hunk 1]
```diff
@@ -130,7 +130,7 @@ const AppRouter = app.getRouter();
 
 const routes = (
   <FlatRoutes>
-    <Navigate key="/" to="catalog" />
+    <Route path="/" element={<Navigate to="catalog" />} />
```
Comment (jhaals) [inline packages/app/src/App.tsx:133]: `/` okay here?

Comment (Rugvip) [reply packages/app/src/App.tsx:133]: Yep it's referring to the absolute path "/", as in only the root of the app

**Pull request closed without merging**